addons/download-button/userscript.js

```diff
@@ -0,0 +1,51 @@
+import downloadBlob from "../../libraries/common/cs/download-blob.js";
+
+export default async function ({ addon, console, msg }) {
+  const vm = addon.tab.traps.vm;
+  const { redux } = addon.tab;
+
+  const username = await addon.auth.fetchUsername();
+
+  await redux.waitForState((state) => state.preview.status.project === "FETCHED");
+  const projectAuthor = redux.state.preview.projectInfo.author?.username;
+
+  const isOwn = username === projectAuthor;
+  const shared = addon.tab.redux.state.preview.projectInfo.is_published;
+
+  async function download() {
+    const downloadButton = document.querySelector(".sa-download-button");
+    downloadButton.classList.add("loading");
+    try {
+      const project = await vm.saveProjectSb3();
+      const title = isOwn ? document.querySelector(".project-title input") : document.querySelector(".project-title");
+      downloadBlob(`${isOwn ? title.value : title.innerText}.sb3`, project);
+    } finally {
+      downloadButton.classList.remove("loading");
+    }
+  }
+
+  const downloadButton = document.createElement("button");
+  downloadButton.innerText = msg("download");
+  downloadButton.onclick = download;
+  downloadButton.classList = "button action-button sa-download-button waiting";
+  downloadButton.id = "sa-download-button";
+
+  function addbutton() {
+    addon.tab.waitForElement(".flex-row .subactions", { markAsSeen: true });
```

Comment:
This line does nothing, you need to `await` it.

Reply:
Solved by my commit https://github.com/ScratchAddons/ScratchAddons/pull/6931/commits/98a00b400f52af9952c518c1cc32af4b8b828c9e